inventory-manager.lic

```diff
@@ -0,0 +1,125 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#invdb
+=end
+
+class InventoryManager
+  include DRC
+  
+  def initialize
+    @active_character = Char.name.to_sym
+    @item_db_path = "data/inventory.yaml"
+
+    arg_definitions = [
+			[
+        { name: 'save', regex: /save/i, description: "Save new or update current character's items to the database." }
+			],
+      [
+        { name: 'search', regex: /search/i, description: 'Start a search across all characters for specified item.' },
+				{ name: 'item', regex: /\w+/i, optional: false, description: 'Item to find.' }
+			],
+			[
+				{ name: 'list', regex: /list/i, description: "Print out a list of names in the database." },
+        { name: 'name', regex: /\w+/i, optional: true, description: "List full inventory for specified character. (Can spam!)" }
+			],
+			[
+				{ name: 'remove', regex: /remove/i, description: "Remove character's inventory." },
+				{ name: 'name', regex: /\w+/i, optional: false, description: 'Character to remove.' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    pause
+
+    setup 
+
+	  if args.save
+		  add_current_inv
+	  elsif args.remove
+      remove_character_data(args.name)
+	  elsif args.search
+      search_for_item(args.item)
+    elsif args.list
+      list_character_inv(args.name)
+    else
+			message 'Type ;invdb help for a usage guide'
+    end
+  end
+
+  def setup
+    @item_data = {}
+
+    if !File.exists?(@item_db_path)
+      File.open(@item_db_path, "w") {}
+      @item_data = OpenStruct.new(YAML.load_file(@item_db_path)).to_h
+    end
+
+    if !File.exists?(@item_db_path)
+      message "Something very wrong is occuring. You don't have a file to open and I can't create one!"
+      message "item-db saves the file to: /lich/scripts/itemDB/inventory.yaml"
+      exit
+    end
+  
+    @item_data = OpenStruct.new(YAML.load_file(@item_db_path)).to_h
+  end
+
+  def add_current_inv
+    @item_data[@active_character] = []
+
+    bput('inv list', 'You have:')
+    while line = get
+      break if line =~ /Roundtime/i
+
+      item = line.lstrip
+      if item[0].eql? '-'
+        item[0] = ''
+        item.concat(' (worn)')
+      else 
+        item.concat(' (inside container)')
+      end
+      @item_data[@active_character] << item
+    end
+
+    message "Saving data for #{@active_character}!"
+    File.open(@item_db_path, 'w') { |file| file.write(@item_data.to_yaml) }
+  end
+
+  def list_character_inv(name = '')
```

Comment:
💡 I'm not a fan of a default arg here, it should always be provided

Reply:
That's a leftover from a previous iteration. I will make that change.